client/client.go

```diff
@@ -125,7 +125,12 @@ func decodeHexData(data string) (*big.Int, error) {
 	const base = 16
 	decoded, ok := new(big.Int).SetString(data[2:], base)
 	if !ok {
-		return nil, fmt.Errorf("could not extract data from %s", data)
+		if data == "0x" {
```

Comment:
that is not what I want.
so if our data length is just 2, then return with big.Int(0) early, before
```
new(big.Int).SetString(data[2:], base)
```

Reply:
Yeah, that's better, handle corner case at the beginning to save one more operation for `decoded, ok := new(big.Int).SetString(data[2:], base)`

Reduce time complexity of O(decoding)